module/VuFind/src/VuFind/Bootstrapper.php

```diff
@@ -394,8 +394,21 @@ protected function initRateLimiter(): void
             $result = $rateLimiterManager->check($event);
             if (!$result['allow']) {
                 $response = $event->getResponse();
-                $response->setStatusCode(429);
-                $response->setContent($result['message']);
+                if ($result['performTurnstileChallenge'] ?? false) {
+                    $response->setStatusCode(307);
+                    $policyId = $rateLimiterManager->checkPolicyUsesTurnstile($event);
+                    $context = base64_encode(json_encode([
+                        'policyId' => $policyId,
+                        'destination' => $event->getRequest()->getUri()->getPath(),
```

Comment:
I could just display the challenge right now.  I'm redirecting to a `/Turnstile/Challenge` URL so the requested page is not accessible to Turnstile's javascript code, so it can't be sent to Cloudflare as part of the signal.  Combining these params and base64-encoding them is an attempt at further obfuscation in case it uses the query parameters as signals as well.  Of course it's still decode-able, but it seemed worthwhile.

Reply:
Might be worth putting some of this reasoning into a comment in the code for future reference... and/or making a separate `redirectToTurnstile` method so it's easy to override/customize the behavior if desired.